diff --git a/doc/html/vortraege.php b/doc/html/vortraege.php
index 2e25017..c515f5e 100644
--- a/doc/html/vortraege.php
+++ b/doc/html/vortraege.php
@@ -14,7 +14,9 @@
 <?php
 	require_once($DOCUMENT_ROOT."/html/php/navbar.php");
 ?>
-<div class="tabltxt-dl">Downloads Vortr&auml;ge:&nbsp;
+<div class="tabltxt-dl">
+<a id='podcast' href='/html/podcast.php'><img src='/images/pod.gif' /> Abonnieren Sie die Vortr&auml;ge als Podcast</a><br>
+Downloads Vortr&auml;ge:&nbsp;
 <?php
 	($result = mysql_query("select sum(downloads) as sumd from vortrag"));
  	$values = mysql_fetch_array($result);
@@ -28,8 +30,7 @@
 	$total_audio_downloads = $values['audiosumd'];
 	echo number_format($total_audio_downloads,0,".","'");
 ?>
-&nbsp;(als Audio-File)&nbsp;
-<br><a id='podcast' href='/html/podcast.php'>Abonnieren Sie die Vortr&auml;ge als Podcast <img src='/images/pod.gif' /></a>
+&nbsp;(als Audio-File)
 </div>
 <table>
 <?php 
