diff --git a/doc/html/vortraege.php b/doc/html/vortraege.php
index c515f5e..4d065d9 100644
--- a/doc/html/vortraege.php
+++ b/doc/html/vortraege.php
@@ -264,11 +264,11 @@ mysql_free_result($result);
 				target='_blank'><img src='../html/images/adobe.gif' alt='PDF File'></a>";
 			}
 			echo "</td>";
-			echo "<td class='tabltxt-c".$suffix."'>";
+			echo "<td class='tabltxt-l".$suffix."'>";
 			if(empty($values["audiofile"])) {
 				echo "&nbsp;";
 			} else {
-				$sizeMb = strval(round($values["audiofile_size"]/1024/1024, 0))."M";
+				$sizeMb = strval(round($values["audiofile_size"]/1024/1024, 2))."M";
 				echo "<a class='pdf-ico".$suffix."'	href='/html/php/download_vortrag.php?id=".$values["id"]."&download=audiofile'
 				target='_blank'><img src='../html/images/audiofile.gif' alt='Audio File'>".$sizeMb."</a>";
 			}
diff --git a/doc/wsadmin/php/vortrag.php b/doc/wsadmin/php/vortrag.php
index d1bd25b..00d8079 100644
--- a/doc/wsadmin/php/vortrag.php
+++ b/doc/wsadmin/php/vortrag.php
@@ -66,28 +66,6 @@ function check_form(form_id) {
 		node.style.color = 'red';
 		alert('Bitte geben Sie einen Titel an.');
 		return false;
-	} else {
-		var node = dojo.byId('patience-bg'); 
-		var inner_node = dojo.byId('patience');
-		var bar = dojo.byId('bar');
-		var mover = dojo.byId('mover');
-		dojo.style.setStyle(node, 'display', 'block');
-		dojo.style.setOpacity(node, 0),
-		dojo.fx.html.fade(node, 1000, 0, 0.5);
-		dojo.style.setStyle(inner_node, 'display', 'block');
-		dojo.style.setStyle(bar, 'display', 'block');
-		dojo.style.setStyle(mover, 'display', 'block');
-		slideToLeft(mover);
-		dojo.io.bind({
-			url: form.action,
-			transport: "IframeTransport",
-			formNode: form,
-			load: function(type, data, event) {
-				document.body.innerHTML = data;
-			},
-			mimetype: "text/html"
-		});
-		return false;
 	}
 }
 //-->
