diff --git a/doc/html/vortraege.php b/doc/html/vortraege.php
index e4e2baf..d49caf5 100644
--- a/doc/html/vortraege.php
+++ b/doc/html/vortraege.php
@@ -162,7 +162,7 @@ mysql_free_result($result);
 		{
 			$thema_condition = "thema_id!=255";
 		}
-		$query = "select Titel, thema, pdf, audiofile, audiofile_length, audiofile_size, audiofile_downloads, id, downloads, zeit, Zielpublikum, location, date_format(gehalten,'%d.%m.%Y') 
+		$query = "select Titel, thema, pdf, audiofile, audiofile_size, audiofile_downloads, id, downloads, zeit, Zielpublikum, location, date_format(gehalten,'%d.%m.%Y') 
 							as gehalten_formatted 
 							from vortrag, thema
 							where $thema_condition and id_thema=thema_id
diff --git a/doc/wsadmin/php/save.php b/doc/wsadmin/php/save.php
index 6d9bb00..16f81a1 100644
--- a/doc/wsadmin/php/save.php
+++ b/doc/wsadmin/php/save.php
@@ -81,9 +81,9 @@ if ($page == "vortrag" && $new == "true"
 
 	$gehalten = input2date($datum);
 
-	$felder_mysql = "Titel, Zusammenfassung, Zielpublikum, gehalten, zeit, location, pdf, audiofile, audiofile_length, audiofile_size, Arbeit, Erziehung, Gesundheit, Familie, thema_id, datumchange";
+	$felder_mysql = "Titel, Zusammenfassung, Zielpublikum, gehalten, zeit, location, pdf, audiofile, audiofile_size, Arbeit, Erziehung, Gesundheit, Familie, thema_id, datumchange";
 
-	$felder_form = "'$Titel', '$Zusammenfassung', '$Zielpublikum', '$gehalten', '$time', '$location', '$file_name', '$audiofile_name', '$audiofile_length', '$audiofile_size', '$Arbeit', '$Erziehung', '$Gesundheit', '$Familie', '$searchnew', '$datumchange'";
+	$felder_form = "'$Titel', '$Zusammenfassung', '$Zielpublikum', '$gehalten', '$time', '$location', '$file_name', '$audiofile_name', '$audiofile_size', '$Arbeit', '$Erziehung', '$Gesundheit', '$Familie', '$searchnew', '$datumchange'";
 
 	if ($file != ""){
 		$path = "../../pdf/" .$file_name;
@@ -115,8 +115,7 @@ if ($page == "vortrag" && $change == "tr
 				gehalten='$gehalten',
 				zeit='$time', 
 				location='$location', 
-				audiofile_name='$audiofile_name',
-				audiofile_length='$audiofile_length',
+				audiofile='$audiofile_name',
 				audiofile_size='$audiofile_size',
 				Arbeit='$Arbeit', 
 				Erziehung='$Erziehung', 
diff --git a/doc/wsadmin/php/vortrag.php b/doc/wsadmin/php/vortrag.php
index c24109a..d1bd25b 100644
--- a/doc/wsadmin/php/vortrag.php
+++ b/doc/wsadmin/php/vortrag.php
@@ -15,7 +15,6 @@ $result = mysql_query ("SELECT * FROM vo
 		$hits = $row["hits"];
 		$file_name = $row["pdf"];
 		$audiofile_name = $row["audiofile"];
-		$audiofile_length = stripslashes(urldecode($row["audiofile_length"]));
 		$audiofile_size = stripslashes(urldecode($row["audiofile_size"]));
 		$filecheck = $DOCUMENT_ROOT."/pdf/$file_name";
 		$audiofilecheck = $DOCUMENT_ROOT."/audio/$audiofile_name";
@@ -256,13 +255,6 @@ if ($change == "true"){
    </td>
   </tr>
   <tr> 
-   <td id='title'>Audiofile L&auml;nge (in Sekunden):</td>
-   <td colspan="2"> 
-	
-<input class='INPUTtext' type="text" name="audiofile_length"  value='<?php print $audiofile_length ?>' size="35" maxlength="100">
-   </td>
-  </tr>
-  <tr> 
    <td id='title'>Audiofile Gr&ouml;sse (in Bytes):</td>
    <td colspan="2"> 
 	
