diff --git a/doc/html/php/download_vortrag.php b/doc/html/php/download_vortrag.php
index f7487ef..55080e1 100644
--- a/doc/html/php/download_vortrag.php
+++ b/doc/html/php/download_vortrag.php
@@ -2,7 +2,7 @@
 
 require_once($DOCUMENT_ROOT."/html/php/mysql_header.php");
 
-if(isset($_GET["id"]) && preg_match("/^[0-9]+$/",$_GET["id"]))
+if(isset($_GET["id"]) && preg_match("/^[0-9]+$/",$_GET["id"]) && $_GET["download"]=='pdf')
 {
   $query = "select pdf from vortrag where id=".$_GET["id"];
   $result = mysql_query($query);
@@ -11,6 +11,14 @@ if(isset($_GET["id"]) && preg_match("/^[
   mysql_query($query);
 
   header("Location: /pdf/".$values["pdf"]);
+} elseif (isset($_GET["id"]) && preg_match("/^[0-9]+$/",$_GET["id"]) && $_GET["download"]=='audiofile') {
+  $query = "select audiofile from vortrag where id=".$_GET["id"];
+  $result = mysql_query($query);
+  $values = mysql_fetch_assoc($result);
+  $query = "update vortrag set audiofile_downloads=downloads+1 where id=".$_GET["id"];
+  mysql_query($query);
+
+  header("Location: /audio/".$values["audiofile"]);
 }
 
 ?>
diff --git a/doc/html/vortraege.php b/doc/html/vortraege.php
index e240894..8450abc 100644
--- a/doc/html/vortraege.php
+++ b/doc/html/vortraege.php
@@ -21,6 +21,14 @@
 	$total_downloads = $values['sumd'];
 	echo number_format($total_downloads,0,".","'");
 ?>
+&nbsp;(als PDF)&nbsp;
+<?php
+	($result = mysql_query("select sum(audiofile_downloads) as audiosumd from vortrag"));
+ 	$values = mysql_fetch_array($result);
+	$total_audio_downloads = $values['audiosumd'];
+	echo number_format($total_audio_downloads,0,".","'");
+?>
+&nbsp;(als Audio-File)&nbsp;
 </div>
 <table>
 <?php 
@@ -31,6 +39,7 @@ $valid = array
 	"Thema"			=>	"thema",
 	"Gelesen"		=>	"hits",
 	"Downloads"	=>	"downloads",
+	"AudioFileDownloads"	=>	"audiofile_downloads",
 	"Gehalten"	=>	"gehalten",
 );
 if(isset($_GET["orderby"]) && isset($valid[$_GET["orderby"]]))
@@ -66,7 +75,7 @@ if($orderdir == "asc")
 	<tr>
 <!--	<td class="tabltxt-r-bg" colspan="6">&Uuml;bersicht Themen:</td>-->
 <td ID= "SiteTitel">Ursula Davatz - Vortr&auml;ge&nbsp; </td>
-<td colspan="6" class="tabltxt-r">
+<td colspan="8" class="tabltxt-r">
 <?php //getmenu.php
 
 if ($new == "true" || $change == "true"){
@@ -133,8 +142,12 @@ mysql_free_result($result);
 	</th>
 	<th>Pdf<br>
 	</th>
+	<th>Audio-File<br>
+	</th>
 	<th><a class="th" href="<?php echo	$url."?orderby=Downloads&amp;orderdir=".$directions["downloads"];?>">Downloads</a><br>
 	</th>
+	<th><a class="th" href="<?php echo	$url."?orderby=AudioFileDownloads&amp;orderdir=".$directions["audiofile_downloads"];?>">Audio File Downloads</a><br>
+	</th>
 	<th><a class="th" href="<?php echo	$url."?orderby=Gehalten&amp;orderdir=".$directions["gehalten"];?>">Gehalten</a><br>
 	</th>
 	</tr>
@@ -149,7 +162,7 @@ mysql_free_result($result);
 		{
 			$thema_condition = "thema_id!=255";
 		}
-		$query = "select Titel, thema, pdf, id, downloads, zeit, Zielpublikum, location, date_format(gehalten,'%d.%m.%Y') 
+		$query = "select Titel, thema, pdf, audiofile, audiofile_length, audiofile_downloads, id, downloads, zeit, Zielpublikum, location, date_format(gehalten,'%d.%m.%Y') 
 							as gehalten_formatted 
 							from vortrag, thema
 							where $thema_condition and id_thema=thema_id
@@ -245,11 +258,20 @@ mysql_free_result($result);
 			}
 			else
 			{
-				echo "<a class='pdf-ico".$suffix."'	href='/html/php/download_vortrag.php?id=".$values["id"]."'
+				echo "<a class='pdf-ico".$suffix."'	href='/html/php/download_vortrag.php?id=".$values["id"]."&download=pdf'
 				target='_blank'><img src='../html/images/adobe.gif' alt='PDF File'></a>";
 			}
 			echo "</td>";
+			echo "<td class='tabltxt-c".$suffix."'>";
+			if(empty($values["audiofile"])) {
+				echo "&nbsp;";
+			} else {
+				echo "<a class='pdf-ico".$suffix."'	href='/html/php/download_vortrag.php?id=".$values["id"]."&download=audiofile'
+				target='_blank'><img src='../html/images/audiofile.gif' alt='Audio File' title='$mbSize'></a>";
+			}
+			echo "</td>";
 			echo "<td class='tabltxt-c".$suffix."'>".urldecode ($values["downloads"])."</td>";	
+			echo "<td class='tabltxt-c".$suffix."'>".urldecode ($values["audiofile_downloads"])."</td>";	
 			echo "<td class='tabltxt-c".$suffix."'>".$values["gehalten_formatted"]."</td>";
 			echo "</tr>\n";
 			if(empty($suffix))
diff --git a/doc/wsadmin/css/browser5.css b/doc/wsadmin/css/browser5.css
index 004e886..4e49035 100644
--- a/doc/wsadmin/css/browser5.css
+++ b/doc/wsadmin/css/browser5.css
@@ -54,7 +54,7 @@ DIV {
    left : 180px; 
    position : absolute; 
    z-index : 1; 
-   visibility : hidden; 
+   visibility : hidden;
 }
 
 
@@ -62,7 +62,7 @@ BODY {
    font : 10pt Arial, Helvetica, sans-serif; 
    color : navy; 
    background-color : #f79231; 
-   background-image : url(../images/swoosh.gif); 
+   background-image : url(../images/swoosh.gif);
    background-repeat : no-repeat; 
    background-attachment : fixed; 
 }
diff --git a/doc/wsadmin/js/standart.js b/doc/wsadmin/js/standart.js
index 236aaea..2842a8c 100644
--- a/doc/wsadmin/js/standart.js
+++ b/doc/wsadmin/js/standart.js
@@ -31,6 +31,17 @@ function pdfDelete(msg){
 		return false;
 	}
 }	
+function audiofileDelete(msg){
+	if (confirm(msg))
+	{
+		document.audiofiledelete.submit();
+		return true;
+	}
+	else
+	{
+		return false;
+	}
+}	
 function popUp(){
 	closeAll();
 	MM_showHideLayers();
diff --git a/doc/wsadmin/php/admin.php b/doc/wsadmin/php/admin.php
index 10adcb6..9d2b00c 100644
--- a/doc/wsadmin/php/admin.php
+++ b/doc/wsadmin/php/admin.php
@@ -55,12 +55,77 @@ header ("Pragma: no-cache");
 <head>
 <title>Ganglion Backend</title>
 <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
+<SCRIPT language="JavaScript" type="text/javascript">
+	djConfig = { 
+		isDebug: true, 
+	};
+</SCRIPT>
+<SCRIPT language="JavaScript" src="/javascript/dojo/dojo.js" type="text/javascript"></SCRIPT>
+<SCRIPT language="JavaScript" type="text/javascript">
+	dojo.require('dojo.fx.*');dojo.require('dojo.fx.html');dojo.require('dojo.io.*');dojo.require('dojo.io.IframeIO');dojo.require('dojo.animation.*');
+</SCRIPT>
 <script language="Javascript" type="text/javascript" src="../js/standart.js" charset="ISO-8859-1">
 </script>
 <?php print $css; ?> 
-
+<style type="text/css">
+div#patience-bg {  
+	display: none;
+	position: absolute; 
+	background-color: black;
+	top: 0px;
+	left: 0px;
+	width: 100%; 
+	height: 100%; 
+	z-index: 1;
+	visibility: visible;
+}
+div#patience {
+	display: none;
+	z-index: 2;
+	visibility: visible;
+	position: absolute;
+	border: red solid 2pt;
+	left: 50%;
+	margin-left: -150px;
+	top: 50%;
+	margin-top: -100px;
+	width: 300px;
+	height: 200px;
+	background-color: white;
+	padding-top: 20px;
+	text-align: center;
+}
+div#bar {
+	display: none;
+	visibility: visible;
+	z-index: 3;
+	position: absolute;
+	border: black solid 1pt;
+	background-color: white;
+	width: 250px;
+	left: 50%;
+	margin-left: -125px;
+	height: 20px;
+	top: 50%;
+	margin-top: -10px;
+}
+div#mover {
+	display: none;
+	visibility: visible;
+	z-index: 3;
+	position: absolute;
+	background-color: red;
+	width: 10px;
+	left: 50%;
+	margin-left: -125px;
+	height: 20px;
+	top: 50%;
+	margin-top: -10px;
+}
+</style>
 </head>
 <body>
+<div id='patience-bg'><div id='patience'>Bitte haben Sie etwas Geduld. <br>Die Dateien werden hochgeladen...<div id='bar'><div id='mover'></div></div></div></div>
 <table width="100%" border="0">
 <tr> 
 <td class='TDSiteTitel' colspan="2">
diff --git a/doc/wsadmin/php/object.php b/doc/wsadmin/php/object.php
index 36dd826..0a97805 100644
--- a/doc/wsadmin/php/object.php
+++ b/doc/wsadmin/php/object.php
@@ -9,6 +9,8 @@ function SendFormButtons($buttons)
 		echo "<input type='button' name='delete' value='Eintrag l&ouml;schen' onclick='entryDelete(titel)'>\n";
 	if (in_array("delete_pdf", $buttons))
 		echo "<input type='button' name='delete' value='Pdf l&ouml;schen' onclick='pdfDelete(pdfname)'>\n";
+	if (in_array("delete_audiofile", $buttons))
+		echo "<input type='button' name='delete' value='Audiofile l&ouml;schen' onclick='audiofileDelete(audiofilename)'>\n";
 }
 function hiddenfields($all){
 	if (isset($new) && $new == "true")
diff --git a/doc/wsadmin/php/save.php b/doc/wsadmin/php/save.php
index 94d7cfc..e1a60ce 100644
--- a/doc/wsadmin/php/save.php
+++ b/doc/wsadmin/php/save.php
@@ -81,27 +81,29 @@ if ($page == "vortrag" && $new == "true"
 
 	$gehalten = input2date($datum);
 
-	$felder_mysql = "Titel, Zusammenfassung, Zielpublikum, gehalten, zeit, location, pdf, Arbeit, Erziehung, Gesundheit, Familie, thema_id, datumchange";
+	$felder_mysql = "Titel, Zusammenfassung, Zielpublikum, gehalten, zeit, location, pdf, audiofile, audiofile_length, Arbeit, Erziehung, Gesundheit, Familie, thema_id, datumchange";
 
-	$felder_form = "'$Titel', '$Zusammenfassung', '$Zielpublikum', '$gehalten', '$time', '$location', '$file_name', '$Arbeit', '$Erziehung', '$Gesundheit', '$Familie', '$searchnew', '$datumchange'";
+	$felder_form = "'$Titel', '$Zusammenfassung', '$Zielpublikum', '$gehalten', '$time', '$location', '$file_name', '$audiofile_name', '$audiofile_length', '$Arbeit', '$Erziehung', '$Gesundheit', '$Familie', '$searchnew', '$datumchange'";
 
 	if ($file != ""){
-
 		$path = "../../pdf/" .$file_name;
-
 		if (!copy($file,$path)) {
-
 			echo "Fehler - kann Datei nicht ablegen<BR>\n";
-
 			exit;
-
 		} else{
-
 			chmod($path, 0777);
-
 		}
-
 	}	
+
+	if ($audiofile != ""){
+		$path = "../../audio/".$audiofile_name;
+		if(!copy($audiofile,$path)) {
+			echo "Fehler - kann Audiofile nicht ablegen<BR>\n";
+			exit;
+		} else {
+			chmod($path, 0777);
+		}
+	}
 	$sql = "INSERT INTO vortrag ($felder_mysql) VALUES ($felder_form)";
 	mysql_query($sql);
 //echo nl2br($query);
@@ -114,73 +116,54 @@ if ($page == "vortrag" && $new == "true"
 //aendern
 
 if ($page == "vortrag" && $change == "true"){
-
 	$datum = "$day.$month.$year";
-
 	$gehalten = input2date($datum);
+	$fields = "	Titel='$Titel', 
+				Zusammenfassung='$Zusammenfassung', 
+				Zielpublikum='$Zielpublikum', 
+				gehalten='$gehalten',
+				zeit='$time', 
+				location='$location', 
+				audiofile_length='$audiofile_length',
+				Arbeit='$Arbeit', 
+				Erziehung='$Erziehung', 
+				Gesundheit='$Gesundheit', 
+				Familie='$Familie', 
+				thema_id='$searchnew', 
+				datumchange='$datumchange'";
 
 	if ($file != ""){
-		//print '->'.$file.'<-';
-
-		$fields = "	Titel='$Titel', 
-					Zusammenfassung='$Zusammenfassung', 
-					Zielpublikum='$Zielpublikum', 
-					gehalten='$gehalten',
-					zeit='$time', 
-					location='$location', 
-					pdf='$file_name', 
-					Arbeit='$Arbeit', 
-					Erziehung='$Erziehung', 
-					Gesundheit='$Gesundheit', 
-					Familie='$Familie', 
-					thema_id='$searchnew', 
-					datumchange='$datumchange'";
-
-
+		$fields = $fields.", pdf='$file_name'";
 		$path = "../../pdf/".$file_name;
-	//	print $Arbeit;
-
 		if (!copy($file,$path)) {
-
 			echo "Fehler - kann Datei nicht ablegen<BR>\n";
-
 			exit;
-
 		} else {
-
 			chmod($path, 0777);
-
 			$delfile = "../../pdf/$oldfile";
-
 			if ($file_name != $oldfile)
-
 				@unlink($delfile);
-
 		}
+	} 
+	if ($audiofile != ""){
+		$fields = $fields.", audiofile='$audiofile_name'";
+		$path = "../../audio/".$audiofile_name;
+		if (!copy($audiofile,$path)) {
+			echo "Fehler - kann Datei nicht ablegen<BR>\n";
+			exit;
+		} else {
+			chmod($path, 0777);
+			$delfile = "../../pdf/$oldaudiofile";
+			if ($audiofile_name != $oldaudiofile)
+				@unlink($delfile);
+		}
+	} 
 
-	} else {
 
-		$fields = "	Titel='$Titel', 
-					Zusammenfassung='$Zusammenfassung', 
-					Zielpublikum='$Zielpublikum', 
-					gehalten='$gehalten', 
-					zeit='$time', 
-					location='$location', 
-					pdf='$oldfile', 
-					Arbeit='$Arbeit', 
-					Erziehung='$Erziehung', 
-					Gesundheit='$Gesundheit', 
-					Familie='$Familie', 
-					thema_id='$searchnew', 
-					datumchange='$datumchange'";
-	}
 
-	
 	$query="UPDATE vortrag SET $fields WHERE id='$id'";
 	mysql_query($query);
-
 	header("Location: admin.php?page=$page&search=$searchnew");
-
 }
 
 //loeschen
@@ -207,6 +190,13 @@ if ($page == "vortrag" && $pdfdelete == 
 
 }
 
+if ($page == "vortrag" && $audiofiledelete == "true"){
+	mysql_query("UPDATE vortrag SET audiofile = '' WHERE id = '$id'");
+	$delfile = "../../audio/$oldaudiofile";
+	@unlink($delfile);
+	@header("Location: admin.php?page=$page&search=$search");
+}
+
 //
 
 //hier werden die Links behandelt
diff --git a/doc/wsadmin/php/uebersicht.php b/doc/wsadmin/php/uebersicht.php
index 5ed3be5..9bdd55e 100644
--- a/doc/wsadmin/php/uebersicht.php
+++ b/doc/wsadmin/php/uebersicht.php
@@ -109,6 +109,7 @@ if ($sortby == "standart" && $search == 
 	elseif	(isset($sortby) && $sortby=="hits")			{	$SortBy="hits DESC";		$order[7] = 1;	}
 	elseif	(isset($sortby) && $sortby=="downloads"	)	{	$SortBy="downloads DESC";	$order[8] = 1;	}
 	elseif	(isset($sortby) && $sortby=="pdf")			{	$SortBy="pdf DESC";			$order[9] = 1;	}
+	elseif	(isset($sortby) && $sortby=="audiofile")			{	$SortBy="audiofile DESC";			$order[9] = 1;	}
 //
 	elseif(isset($sortby) && $sortby=="Titel-")		{	$SortBy="Titel DESC";		}
 	elseif(isset($sortby) && $sortby=="datum-")	{	$SortBy="gehalten ASC";		}
@@ -119,6 +120,7 @@ if ($sortby == "standart" && $search == 
 	elseif	(isset($sortby) && $sortby=="Erziehung-")		{	$SortBy="Erziehung ASC";	}
 	elseif	(isset($sortby) && $sortby=="hits-")			{	$SortBy="hits ASC";			}
 	elseif	(isset($sortby) && $sortby=="pdf-")			{	$SortBy="pdf ASC";			}
+	elseif	(isset($sortby) && $sortby=="audiofile-")			{	$SortBy="audiofile ASC";			}
 //
 //
 	$order[0]==1?$Titel="Titel-"				: $Titel="Titel";
@@ -131,6 +133,7 @@ if ($sortby == "standart" && $search == 
 	$order[7]==1?$hits="hits-"					: $hits="hits";
 	$order[8]==1?$downloads="downloads-"	: $downloads="downloads";
 	$order[9]==1?$pdf="pdf-"				: $pdf="pdf";
+	$order[9]==1?$audiofile="audiofile-"				: $audiofile="audiofile";
 //
 
 	echo "<table class='TABLEliste' width='100%' cellspacing='2'>\n";
@@ -151,6 +154,10 @@ if ($sortby == "standart" && $search == 
 	echo "</th>\n";
 
 	echo "<th class='THzahl'>\n";
+	echo "	<a class='Atitel' href='$url&sortby=$audiofile'\n";	?>	onMouseOut="FnormText()" onMouseOver="StatusMsg('Sortieren nach Audiofile');return document.returnValue">Audiofile</a><?php
+	echo "</th>\n";
+
+	echo "<th class='THzahl'>\n";
 	echo "	<a class='Atitel' href='$url&sortby=$hits'\n";	?>	onMouseOut="FnormText()" onMouseOver="StatusMsg('Sortieren nach gelesen');return document.returnValue">Gelesen</a><?php
 	echo "</th>\n";
 
@@ -216,13 +223,20 @@ if (isset($request) && $request=="uebers
 					$statusTitel = addslashes($Titel);
 					$statusTitel = htmlentities($Titel);
 					$pdf_file = $row["pdf"];
+					$audio_file = $row["audiofile"];
 					$pdf_path = "../../pdf/".$pdf_file;
+					$audio_path = "../../audio/".$audio_file;
 					if (@filetype($pdf_path) == "file"){
 						$pdf_file = $ok;
 					}
 					else{
 						$pdf_file ="";
-						}
+					}
+					if(@filetype($audio_path) == "file") {
+						$audio_file = $ok;
+					} else { 
+						$audio_file ="";
+					}
 					$thema = stripslashes(urldecode($row["thema"]));
 					$thema = htmlentities($thema);
 					$hits = $row["hits"];
@@ -244,6 +258,7 @@ if (isset($request) && $request=="uebers
 					echo "		<a href='$url&change=true&id=$id&thema=$thema_id'";?> onMouseOut="FnormText()" onMouseOver="StatusMsg('<?php echo "Detaileintrag: $statusTitel";?>');return document.returnValue"> <?php echo"$Titel</a></td>\n";
 					echo "	<td class='TDdatum'>".$gehalten."</td>\n";
 					echo "	<td class='TDzahl'>".$pdf_file."</td>\n";
+					echo "	<td class='TDzahl'>".$audio_file."</td>\n";
 					echo "	<td class='TDzahl'>".$row["hits"]."</td>\n";
 					echo "	<td class='TDzahl'>".$row["downloads"]."</td>\n";
 					echo "	<td class='TDthema'>$thema</td>\n";
diff --git a/doc/wsadmin/php/vortrag.php b/doc/wsadmin/php/vortrag.php
index 19f140b..c0c7e7a 100644
--- a/doc/wsadmin/php/vortrag.php
+++ b/doc/wsadmin/php/vortrag.php
@@ -14,7 +14,10 @@ $result = mysql_query ("SELECT * FROM vo
 		$location = stripslashes(urldecode($row["location"]));
 		$hits = $row["hits"];
 		$file_name = $row["pdf"];
+		$audiofile_name = $row["audiofile"];
+		$audiofile_length = stripslashes(urldecode($row["audiofile_length"]));
 		$filecheck = $DOCUMENT_ROOT."/pdf/$file_name";
+		$audiofilecheck = $DOCUMENT_ROOT."/audio/$audiofile_name";
 		$Familie = $row["Familie"];      
 		$Arbeit = $row["Arbeit"]; 
 		$Gesundheit = $row["Gesundheit"];   
@@ -28,6 +31,7 @@ $datum = datumsplitt($datum);
 $datumchange = date("Y-m-d");
 $msgConf = "Wollen Sie den Eintrag: <$javatitel> wirklich löschen?";
 $msgConf_pdf = "Wollen Sie diese Pdf Datei: <$file_name> wirklich löschen?";
+$msgConf_audiofile = "Wollen Sie dieses Audiofile: <$audiofile_name> wirklich löschen?";
 }
 elseif ($new == "true") {
 $datumchange = date("Y-m-d");
@@ -42,18 +46,54 @@ $minute = -1;
 <?php
 echo "var titel = '$msgConf';\n";
 echo "var pdfname = '$msgConf_pdf';\n";
+echo "var audiofilename = '$msgConf_audiofile';\n";
 ?>
-function check_form(form)
-{
-if(form.Titel.value=='')
-	{
-	alert('Bitte geben Sie einen Titel ein.');
-	return false;
+function slideToRight(mover) {
+	var callback = function() {
+		slideToLeft(mover);
+	}
+	dojo.fx.html.slide(mover, 20000, [365,10], [125,10], callback);
+}
+function slideToLeft(mover) {
+	var callback = function() {
+		slideToRight(mover);
+	}
+	dojo.fx.html.slide(mover, 20000, [125,10], [365,10], callback);
+}
+function check_form(form_id) {
+	var form = dojo.byId(form_id);
+	if(form.Titel.value=='') {
+		var node = dojo.byId('title');
+		node.style.color = 'red';
+		alert('Bitte geben Sie einen Titel an.');
+		return false;
+	} else {
+		var node = dojo.byId('patience-bg'); 
+		var inner_node = dojo.byId('patience');
+		var bar = dojo.byId('bar');
+		var mover = dojo.byId('mover');
+		dojo.style.setStyle(node, 'display', 'block');
+		dojo.style.setOpacity(node, 0),
+		dojo.fx.html.fade(node, 1000, 0, 0.5);
+		dojo.style.setStyle(inner_node, 'display', 'block');
+		dojo.style.setStyle(bar, 'display', 'block');
+		dojo.style.setStyle(mover, 'display', 'block');
+		slideToLeft(mover);
+		dojo.io.bind({
+			url: form.action,
+			transport: "IframeTransport",
+			formNode: form,
+			load: function(type, data, event) {
+				document.body.innerHTML = data;
+			},
+			mimetype: "text/html"
+		});
+		return false;
 	}
 }
 //-->
 </script>
-<form method="post" action="save.php" name="vortrag" onSubmit='return check_form(this);' enctype="multipart/form-data">
+<form method="post" id='lectureForm' action="save.php" name="vortrag" onSubmit='return check_form("lectureForm");' enctype="multipart/form-data">
 
  
 <table class="TABLEvortrag">
@@ -174,7 +214,7 @@ for($i=0; $i < 60; $i+=15){
    </td>
   </tr>
   <tr> 
-   <td>Titel:</td>
+   <td id='title'>Titel:</td>
    <td colspan="2"> 
 	
 <input class='INPUTtext' type="text" name="Titel"  value='<?php print $Titel ?>' size="35" maxlength="100">
@@ -208,6 +248,34 @@ if ($change == "true"){
 	echo "</tr>\n";
 }
 ?> 
+	<tr>
+		<td>Audiofile:</td>
+		<td colspan="2">
+			<input class='INPUTtext' type='file' name='audiofile' />
+		</td>
+	</tr>
+  <?php
+if ($change == "true"){
+	echo "<tr>\n";
+	echo "<td>&nbsp;</td>\n";
+	if (@filetype($audiofilecheck) == "file")	
+	{
+		echo "<td>Es existiert ein Audiofile: $audiofile_name</td>\n"; 
+	} 
+	else	
+	{ 
+		echo "<td>Es existiert kein Audiofile</td>"; 
+	}
+	echo "</tr>\n";
+}
+?> 
+  <tr> 
+   <td id='title'>Audiofile L&auml;nge (in Sekunden):</td>
+   <td colspan="2"> 
+	
+<input class='INPUTtext' type="text" name="audiofile_length"  value='<?php print $audiofile_length ?>' size="35" maxlength="100">
+   </td>
+  </tr>
   <tr> 
    <td rowspan="4">Bereich:</td>
    <td> 
@@ -245,10 +313,11 @@ if (isset($change) && $change == "true")
 	echo "<input type='hidden' name='change' value='true'>\n";
 	echo "<input type='hidden' name='id' value='$id'>\n";
 	echo "<input type='hidden' name='oldfile' value='$file_name'>\n";
+	echo "<input type='hidden' name='oldaudiofile' value='$audiofile_name'>\n";
 
 }
 ?> </td>
-  <td colspan="2"> <?php SendFormButtons(array("delete", "delete_pdf")); ?> </td>
+  <td colspan="2"> <?php SendFormButtons(array("delete", "delete_pdf", "delete_audiofile")); ?> </td>
   </tr>
  </table>
 </form>
@@ -258,6 +327,7 @@ if (isset($change) && $change == "true")
 <input type="hidden" name="search" value="<?php echo $search ?>">
 <input type="hidden" name="id" value="<?php echo $id ?>">
 <input type="hidden" name="oldfile" value="<?php echo $file_name ?>">
+<input type="hidden" name="oldaudiofile" value="<?php echo $audiofile_name ?>">
 </form>
 <form method="post" action="save.php" name="pdfdelete" enctype="multipart/form-data">
 <input type="hidden" name="page" value="<?php echo $page ?>">
@@ -266,3 +336,10 @@ if (isset($change) && $change == "true")
 <input type="hidden" name="id" value="<?php echo $id ?>">
 <input type="hidden" name="oldfile" value="<?php echo $file_name ?>">
 </form>
+<form method="post" action="save.php" name="audiofiledelete" enctype="multipart/form-data">
+<input type="hidden" name="page" value="<?php echo $page ?>">
+<input type="hidden" name="audiofiledelete" value="true">
+<input type="hidden" name="search" value="<?php echo $search ?>">
+<input type="hidden" name="id" value="<?php echo $id ?>">
+<input type="hidden" name="oldaudiofile" value="<?php echo $audiofile_name ?>">
+</form>
diff --git a/doc/wsadmin/webmaster.html b/doc/wsadmin/webmaster.html
index 4c8c030..183510a 100644
--- a/doc/wsadmin/webmaster.html
+++ b/doc/wsadmin/webmaster.html
@@ -23,4 +23,4 @@ MM_reloadPage(true);
 </frameset>
 <noframes>
 </noframes>
-</html>
\ No newline at end of file
+</html>
