--- printVortraege.php	2013-06-25 15:06:19.443448484 +0200
+++ printVortraege.php	2013-06-26 07:58:01.578401596 +0200
@@ -55,17 +55,17 @@
 	elseif	(isset($sortby) && $sortby=="hits-")			{	$SortBy="hits ASC";			}
 	elseif	(isset($sortby) && $sortby=="pdf-")			{	$SortBy="pdf ASC";			}
 //
- //
-	$order[0]==1?$Titel="Titel-"				: $Titel="Titel";
-	$order[1]==1?$datum="datum-"			: $datum="datum";
-	$order[2]==1?$Thema="Thema-"			: $Thema="Thema";
-	$order[3]==1?$Familie="Familie-"			: $Familie="Familie";
-	$order[4]==1?$Arbeit="Arbeit-"			: $Arbeit="Arbeit";
-	$order[5]==1?$Gesundheit="Gesundheit-"	: $Gesundheit="Gesundheit";
-	$order[6]==1?$Erziehung="Erziehung-"		: $Erziehung="Erziehung";
-	$order[7]==1?$hits="hits-"					: $hits="hits";
-	$order[8]==1?$downloads="downloads-"	: $downloads="downloads";
-	$order[9]==1?$pdf="pdf-"				: $pdf="pdf";
+//
+(isset($order[0]) && $order[0] == 1) ? $Titel = "Titel-"           : $Titel = "Titel";
+(isset($order[1]) && $order[1] == 1) ? $datum = "datum-"           : $datum = "datum";
+(isset($order[2]) && $order[2] == 1) ? $Thema = "Thema-"           : $Thema = "Thema";
+(isset($order[3]) && $order[3] == 1) ? $Familie = "Familie-"       : $Familie = "Familie";
+(isset($order[4]) && $order[4] == 1) ? $Arbeit = "Arbeit-"         : $Arbeit = "Arbeit";
+(isset($order[5]) && $order[5] == 1) ? $Gesundheit = "Gesundheit-" : $Gesundheit = "Gesundheit";
+(isset($order[6]) && $order[6] == 1) ? $Erziehung = "Erziehung-"   : $Erziehung = "Erziehung";
+(isset($order[7]) && $order[7] == 1) ? $hits = "hits-"             : $hits = "hits";
+(isset($order[8]) && $order[8] == 1) ? $downloads = "downloads-"   : $downloads = "downloads";
+(isset($order[9]) && $order[9] == 1) ? $pdf = "pdf-"               : $pdf = "pdf";
 //
 
 
